From: Gordon Ball <gordon@chronitis.net>
Date: Tue, 15 Sep 2020 12:31:43 +0000
Subject: Patch flaky or unavailable tests

* Don't test ipywidgets (up to date version not available)
* Skip other notebook tests which appear to race
* Extend timeouts for parallel tests
---
 nbclient/tests/test_client.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/nbclient/tests/test_client.py
+++ b/nbclient/tests/test_client.py
@@ -329,8 +329,6 @@
                 "allow_errors": True,
             },
         ),
-        ("JupyterWidgets.ipynb", {"kernel_name": "python"}),
-        ("Skip Exceptions with Cell Tags.ipynb", {"kernel_name": "python"}),
         ("Skip Exceptions.ipynb", {"kernel_name": "python", "allow_errors": True}),
         ("Skip Execution with Cell Tag.ipynb", {"kernel_name": "python"}),
         ("SVG.ipynb", {"kernel_name": "python"}),
@@ -366,7 +364,7 @@
         for t in threads:
             t.start()
         for t in threads:
-            t.join(timeout=2)
+            t.join(timeout=20)
 
     captured = capfd.readouterr()
     assert filter_messages_on_error_output(captured.err) == ""
@@ -852,6 +850,7 @@
         executed = execute(original, os.path.dirname(filename))
         assert_notebooks_equal(original, executed)
 
+    @pytest.mark.skip()
     def test_widgets(self):
         """Runs a test notebook with widgets and checks the widget state is saved."""
         input_file = os.path.join(current_dir, 'files', 'JupyterWidgets.ipynb')
